codalab/server/bundle_manager.py

```
@@ -351,13 +372,19 @@ def _filter_and_sort_workers(workers_list, bundle, bundle_resources):
         Filters the workers to those that can run the given bundle and returns
         the list sorted in order of preference for running the bundle.
         """
-
         # keep track of which workers have GPUs
         has_gpu = {}
         for worker in workers_list:
             worker_id = worker['worker_id']
             has_gpu[worker_id] = worker['gpus'] > 0
 
+        # Filter by tag.
+        request_queue = bundle.metadata.request_queue
+        if request_queue:
+            tagm = re.match('tag=(.+)', request_queue)
+            if tagm:
+                workers_list = [worker for worker in workers_list if worker['tag'] == tagm.group(1)]
```

We still need the `else return []` statement here. We don't want to silently ignore malformed request queue arguments.